database/ServiceMySql.php

```diff
@@ -8,87 +8,127 @@
 
 class ServiceMySql implements ServiceInterface
 {
-	private $conn;
-
-	public function __construct(\PDO $conn)
-	{
-		$this->conn = $conn;
-	}
-
-	public function getAll()
-	{
-		try {
-			$sql = $this->conn->prepare("SELECT * FROM guest.student");
-			$sql->execute();
-			$rows = $sql->fetchAll();
-			if($rows == null){
-				throw new EmptyTableException("Table is empty");
-			}
-			$var = array();
-			foreach ($rows as $row => $value) {
-				$object = new Student();
-				$var[] = $object
-					->setId($value['id'])
-					->setName($value['name'])
-					->setSurname($value['surname'])
-					->setIndexNo($value['indexno'])
-					->setAddress($value['address']);
-			}
-
-			return $var;
-		} catch (EmptyTableException $e){
-			echo $e->getMessage();
-		}
-	}
-
-	public function getOne($id)
-	{
-		try {
-			$sql = $this->conn->prepare("SELECT * FROM guest.student WHERE id=" . $id . " LIMIT 1");
-			$sql->execute();
-			$row = $sql->fetch();
-			if($row == null){
-				throw new InvalidIdException("Invalid id");
-			}
-			$object = new Student();
-			$object
-				->setId($row['id'])
-				->setName($row['name'])
-				->setSurname($row['surname'])
-				->setIndexNo($row['indexno'])
-				->setAddress($row['address']);
-
-			return $object;
-		} catch (InvalidIdException $e){
-			echo $e->getMessage();
-		}
-	}
-
-	public function create()
-	{
-		try {
-			$sql = $this->conn->prepare("INSERT INTO guest.student(name, surname, indexno, address) VALUES (:name, :surname, :indexno, :address)");
-
-			$sql->bindParam(':name', $_POST['name']);
-			$sql->bindParam(':surname', $_POST['surname']);
-			$sql->bindParam(':indexno', $_POST['indexno']);
-			$sql->bindParam(':address', $_POST['address']);
-
-			$sql->execute();
-			echo "Resource successfully created";
-		} catch (\Exception $e){
-			echo "Error creating resource: " . $e->getMessage();
-		}
-	}
-
-	public function delete($id)
-	{
-
-	}
-
-	public function update($id)
-	{
-		// TODO: Implement update() method.
-	}
+    private $conn;
 
+    private final function putValues($id, $data){
+        $putValues = array();
+        $idValues = $this->getOne($id);
```

Comment:
do we really need to call getOne($id) function or?

Reply:
checkId($id) ?

Reply:
yup! we only need to retrieve it from the DB and check it's values. we do not need to instantiate the obj and so on...

Reply:
Already done it, just wait to push